src/Stratis.Features.Collateral/CheckCollateralCommitmentHeightRule.cs

```diff
@@ -15,12 +16,22 @@ namespace Stratis.Features.Collateral
     /// </summary>
     public sealed class CheckCollateralCommitmentHeightRule : FullValidationConsensusRule
     {
+        private readonly int? lastCheckPointHeight;
+
+        public CheckCollateralCommitmentHeightRule(Network network)
+        {
+            this.lastCheckPointHeight = network.Checkpoints.OrderByDescending(c => c.Key).Select(c => c.Key).FirstOrDefault();
+        }
+
         public override Task RunAsync(RuleContext context)
         {
             // The genesis block won't contain any commitment data.
             if (context.ValidationContext.ChainedHeaderToValidate.Height == 0)
                 return Task.CompletedTask;
 
+            if (context.ValidationContext.ChainedHeaderToValidate.Height <= this.lastCheckPointHeight)
```

Comment:
I agree that this rule seems to be safely skippable as it does not change state. However, have you considered the `RuleContext.SkipValidation` flag used elsewhere in the code? e.g. see how it is used in `CheckPowTransactionRule`

Reply:
Done.